devscripts/update-version.py

```diff
@@ -30,30 +32,43 @@ def get_new_version(revision):
 
 def get_git_head():
     with contextlib.suppress(Exception):
-        sp = subprocess.Popen(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE)
-        return sp.communicate()[0].decode().strip() or None
+        return subprocess.check_output(['git', 'rev-parse', 'HEAD'], text=True).strip() or None
 
 
-revision = (sys.argv + [''])[1]
-VERSION = get_new_version(revision)
-GIT_HEAD = get_git_head()
-
-VERSION_FILE = f'''\
+VERSION_TEMPLATE = '''\
 # Autogenerated by devscripts/update-version.py
 
-__version__ = {VERSION!r}
+__version__ = {version!r}
 
-RELEASE_GIT_HEAD = {GIT_HEAD!r}
+RELEASE_GIT_HEAD = {git_head!r}
 
 VARIANT = None
 
 UPDATE_HINT = None
 
-CHANNEL = {"nightly" if revision else "stable"!r}
+CHANNEL = {channel!r}
 '''
 
-write_file('yt_dlp/version.py', VERSION_FILE)
-github_output = os.getenv('GITHUB_OUTPUT')
-if github_output:
-    write_file(github_output, f'ytdlp_version={VERSION}\n', 'a')
-print(f'\nVersion = {VERSION}, Git HEAD = {GIT_HEAD}')
+if __name__=='__main__':
+    parser = argparse.ArgumentParser(description='Update the version.py file')
+    parser.add_argument(
+        '-c', '--channel', choices=['stable', 'nightly'], default='stable',
+        help='Select update channel (default: stable)')
+    parser.add_argument(
+        '-o', '--output', default='yt_dlp/version.py',
+        help='The output file to write to (default: yt_dlp/version.py)')
+    parser.add_argument(
+        'version', nargs='?', default=None,
+        help='A full version, version or revision to override generated ones')
+    args = parser.parse_args()
+
+    git_head = get_git_head()
+    version = (
+        get_new_version(None, None) if not args.version
+        else args.version if args.version.count('.') == 3  # Full version
+        else get_new_version(args.version, None) if args.version.count('.') == 2
+        else get_new_version(None, args.version))  # Revision only
```

Comment:
I think if we are passing version, revision shouldn't be automatically added

```suggestion
        args.version if '.' in args.version
        else get_new_version(None, args.version))
```

Reply:
Even if that could result in a clash? Maybe only if `--force` or something? Or does it not matter?

Reply:
I have done it your way for now. I don't think we need to worry about it too much though